SaucyBot/Site/Instagram.cs

```diff
@@ -0,0 +1,89 @@
+using System.Text.RegularExpressions;
+using Discord;
+using Discord.WebSocket;
+using SaucyBot.Site.Response;
+
+namespace SaucyBot.Site;
+
+public sealed class Instagram : BaseSite
+{
+    public override string Identifier => "Instagram";
+
+    // Match Instagram URLs from instagram.com, www.instagram.com, m.instagram.com, l.instagram.com
+    // Only capture /p/, /reel/, /reels/ paths
+    protected override string Pattern => 
+        @"https?:\/\/(?<host>(?:www\.|m\.|l\.)?instagram\.com)\/(?<path>(?:p|reel|reels)\/[^\/\s\?\#]+(?:\/[^\s\?\#]*)?)(?<query>\?[^\s\#]*)?(?<fragment>\#[^\s]*)?";
+
+    protected override Color Color => new(0xE4405F);
+
+    private readonly ILogger<Instagram> _logger;
+
+    public Instagram(ILogger<Instagram> logger)
+    {
+        _logger = logger;
+    }
+
+    public override async Task<ProcessResponse?> Process(Match match, SocketUserMessage? message = null)
+    {
+        var originalUrl = match.Value;
+        var host = match.Groups["host"].Value.ToLowerInvariant();
+
+        // Idempotent: skip if already rewritten
+        if (host.Contains("kkinstagram.com"))
+        {
+            _logger.LogDebug("Instagram URL already rewritten to kkinstagram.com, skipping: {Url}", originalUrl);
+            return null;
+        }
```

Comment:
The `host` group of the regex match will never contain "kkinstagram.com" as the regex cannot match against it: https://regex101.com/r/tQqzpB/1

Additionally even if it did, this is made redundant by your `IsValidInstagramHost` call below.